pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -157,6 +157,19 @@ func (p *Properties) SetAzureStackCloudSpec() error {
 			p.CustomCloudProfile.AzureEnvironmentSpecConfig = &azureStackCloudSpec
 		}
 		AzureCloudSpecEnvMap[AzureStackCloud] = azureStackCloudSpec
+
+		// Update Fault domain value to 3 for AzureStack Case
+		// for both Master and Agent pools.
+
+		faultDomaincount := 3
```

Comment:
Will these changes break upgrade scenarios for existing Azure Stack clusters? I don't think you can re-deploy an ARM template that has, say, a FD count of 2, and change it to 3.

Reply:
Yes. Good point. Do you have answer how did you handle similar situation ?. 
As if I recall correctly previously faultdomain value was hardcoded to 2 and now its dynamic based on the region (in some region now this value became 3 compared to 2 previously).

Reply:
Now in aks-engine, when you provision a new VMAS cluster it uses that ARM formula that calculates the FD count for the region. When upgrading, the existing FD count is looked up and used, because the count can never be changed after creation.

Reply:
So if this is just setting defaults for an Azure Stack cluster, and the upgrade code follows the same code path, then this change should be fine. New clusters will start using 3 instead of 2 (I assume Azure Stack's VMAS implementation is always ok with 3), and existing clusters will respect 2 if that's how they were provisioned.

Is there some pattern to the `location` field in Azure Stack? It's also possible we could modify the ARM formula to take Azure Stack into account.

Reply:
Thx for the update @mboersma. Looks like on upgrade flow Azure Stack is not looking up existing value. This can cause failure. Let me take a look on Azure Stack upgrade flow.

I looked at the code where ARM formula was getting modify but in Azure Stack based on location we cant say what is the fault domain value. Let me check with team if their is any better way to derive fault domain value.

Reply:
@mboersma/ @jackfrancis : After adding check extra check in updated PR things are working for upgrade scenario as well for existing customer :). 
Please let me know now you are fine with new changes.

@honcao @jadarsie: Please do review and let me know your comments.